api/app/src/routes/middlewares/analytics.ts

```diff
@@ -0,0 +1,33 @@
+import { NextFunction, Request, Response } from "express";
+import url from "url";
+
+import { getCxIdOrFail } from "../util";
+import { ApiTypes } from "../../command/usage/report-usage";
+import { Config } from "../../shared/config";
+import { analytics } from "../../shared/analytics";
+
+export const analyzeRoute = (apiType: ApiTypes) => {
+  return (req: Request, res: Response, next: NextFunction): void => {
+    const cxId = getCxIdOrFail(req);
+
+    // Link route gets converted to /patient/2.16.840.1.113883.3.9621.5.505.2.1000000004/link
+    // Should we maybe consider /link with patientId as a query?
+    const reqUrl = apiType === ApiTypes.devices ? req.baseUrl : url.parse(req.url).pathname;
```

Comment:
Don't we want `.url.pathname` for devices api as well?`baseUrl` gives path where the current router was mounted - [source](https://expressjs.com/en/4x/api.html#req.baseUrl).

Also, what problems the link url creates?

Reply:
If the goal here is to track usage per endpoint, should we capture the routes as such without an explicit ID (if possible)?

`/medical/v1/patient/:id`

Also, would it be useful to instead hook this into the end of every call to capture the HTTP code? That way we can get stats into 200s vs 500s and etc